pkg/templates/pdbmaxunavailable/template_test.go

```diff
@@ -28,11 +28,49 @@ func (p *PDBTestSuite) SetupTest() {
 	p.ctx = mocks.NewMockContext()
 }
 
-func (p *PDBTestSuite) TestPDBMaxUnavailableZero() {
+func (p *PDBTestSuite) TestPDBMaxUnavailableZeroAsInteger() {
```

Comment:
How about converting this tests to table driven tests?
```go
func (p *PDBTestSuite) TestPDB() {
	testCases := []struct {
		Name           string
		Message        string
		MaxUnavailable intstr.IntOrString
	}{
		{
			Name:           "Invalid Value String",
			Message:        "maxUnavailable has invalid value [invalid-value]",
			MaxUnavailable: intstr.FromString("invalid-value"),
		},
		{
			Name:           "Invalid Value String",
			Message:        "maxUnavailable has invalid value [0]",
			MaxUnavailable: intstr.FromString("0"),
		},
		{
			Name:           "One As String Percentage",
			MaxUnavailable: intstr.FromString("1%"),
		},
		{
			Name:           "One As Integer",
			MaxUnavailable: intstr.FromInt(1),
		},
		{
			Name:           "Zero As String Percentage",
			Message:        "MaxUnavailable is set to 0",
			MaxUnavailable: intstr.FromString("0%"),
		},
		{
			Name:           "Zero As Integer",
			Message:        "MaxUnavailable is set to 0",
			MaxUnavailable: intstr.FromInt(0),
		},
	}

	for _, tc := range testCases {
		tc := tc
		p.Run(tc.Name, func() {
			p.ctx.AddMockPodDisruptionBudget(p.T(), "test-pdb")
			p.ctx.ModifyPodDisruptionBudget(p.T(), "test-pdb", func(pdb *pdbV1.PodDisruptionBudget) {
				pdb.Spec.MaxUnavailable = &tc.MaxUnavailable
			})

			expected := map[string][]diagnostic.Diagnostic{}
			if tc.Message != "" {
				expected = map[string][]diagnostic.Diagnostic{
					"test-pdb": {{Message: tc.Message}},
				}
			}

			p.Validate(p.ctx, []templates.TestCase{
				{
					Param:                    params.Params{},
					Diagnostics:              expected,
					ExpectInstantiationError: false,
				},
			})
		})
	}
}
```

Reply:
make sense. done